src/commands/commerce/extension/map.ts

```diff
@@ -0,0 +1,154 @@
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<string> {
+        this.ux.log(`Accessing Store using username: ${this.org.getUsername()} \n..........`);
+
+        return this.processMapExtension(
+            this.flags['registered-extension-name'],
+            this.flags['store-name'],
+            this.flags['store-id'],
+            this.org.getUsername()
+        );
+    }
+
+    public processMapExtension(extensionName: string, storeName: string, storeId: string, userName: string): string {
+        if (storeName === undefined && storeId === undefined) {
+            throw new SfdxError(msgs.getMessage('extension.map.undefinedName'));
+        }
+        const storeid = this.validateStoreId(storeName, storeId, userName);
+        const registeredExternalServiceId = this.getRegisteredExtensionId(extensionName, userName);
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const results = forceDataRecordCreate(
+            'StoreIntegratedService',
+            `Integration=${registeredExternalServiceId} StoreId=${storeid} ServiceProviderType='Extension'`,
+            userName
+        );
+
+        if (results instanceof SfdxError) {
+            throw new SfdxError(msgs.getMessage('extension.map.error', [extensionName, '\n', results.message]));
+        }
+        // JSON response of inserted record
+        return this.getInsertedRecord(storeid, registeredExternalServiceId);
+    }
+
+    private getRegisteredExtensionId(extensionName: string, userName: string): string {
+        let registeredExternalServiceId: string;
+        try {
+            registeredExternalServiceId = forceDataSoql(
+                `SELECT Id FROM RegisteredExternalService WHERE DeveloperName='${extensionName}'`,
+                userName
+            ).result.records[0].Id;
+        } catch (error) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            const errorMsg = msgs.getMessage('extension.map.nonexistent', [extensionName, '\n', error.message]);
+            throw new SfdxError(errorMsg);
+        }
+        return registeredExternalServiceId;
+    }
+
+    private validateStoreId(storeName: string, storeId: string, userName: string): string {
+        let fResult: Result<QueryResult>;
+        let duplicate: boolean;
+        if (storeId === undefined) {
+            try {
+                fResult = forceDataSoql(`SELECT Id FROM WebStore WHERE Name='${storeName}'`, userName);
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                if (fResult.result.totalSize > 1) {
+                    duplicate = true;
+                }
+            } catch (e) {
+                throw new SfdxError(
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                    msgs.getMessage('extension.map.errStoreName', [storeName, '\n', e.message])
```

Comment:
This error message is not being thrown because when you pass a storename that doesn't exist it'll return a query with 0 results rather than throw an error

<img width="640" alt="Screen Shot 2022-08-11 at 2 56 59 PM" src="https://user-images.githubusercontent.com/91205717/184217974-4b68c947-bf6b-4aa6-a349-5af1a02508e9.png">
.